integration-tests/java/src/integration-test/java/com/stellarstation/api/test/satellite/SatelliteStreamerTest.java

```diff
@@ -0,0 +1,156 @@
+                  ByteString data = response.getReceiveTelemetryResponse().getTelemetry().getData();
+                  if (data.size() > 2) {
+                    // The second last byte of the telemetry indicates the current state of the
+                    // fake satellite used in the test. The value is either of 0 or 1.
+                    int state = data.byteAt(data.size() - 2);
+                    safeModeStates.add(state);
+
+                    if (safeModeStates.size() == 2) {
+                      latch.countDown();
+                    }
+
+                    // Send a command to toggle the state.
+                    SendSatelliteCommandsRequest command =
+                        SendSatelliteCommandsRequest.newBuilder()
+                            .addCommand(ByteString.copyFrom(new byte[] {0x01, 0x01}))
+                            .build();
+                    queue.add(command);
+                  }
+                } else {
+                  GroundStationState gsState =
+                      response.getStreamEvent().getPlanMonitoringEvent().getGroundStationState();
+                  AntennaState antennaState = gsState.getAntenna();
+
+                  azimuthCommand.add(antennaState.getAzimuth().getCommand());
+                  azimuthMeasured.add(antennaState.getAzimuth().getMeasured());
+                  elevationCommand.add(antennaState.getElevation().getCommand());
+                  elevationMeasured.add(antennaState.getElevation().getMeasured());
+                }
+              }
+
+              @Override
+              public void onError(Throwable t) {
+                latch.countDown();
+              }
+
+              @Override
+              public void onCompleted() {
+                latch.countDown();
+              }
+            });
+
+    // Stop tests after 30 seconds.
+    Executor executor = Executors.newSingleThreadExecutor();
+    executor.execute(
+        () -> {
+          try {
+            latch.await(30, TimeUnit.SECONDS);
+            latch.countDown();
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        });
+
+    // Sends commands in the blocking queue to the API.
+    requestObserver.onNext(
+        SatelliteStreamRequest.newBuilder().setSatelliteId(SATELLITE_ID).build());
+
+    while (latch.getCount() > 0) {
+      SendSatelliteCommandsRequest commandsRequest = queue.take();
+      requestObserver.onNext(
+          SatelliteStreamRequest.newBuilder()
+              .setSatelliteId(SATELLITE_ID)
+              .setSendSatelliteCommandsRequest(commandsRequest)
+              .build());
+    }
+
+    // Check safe mode state is changed..
+    assertThat(safeModeStates).hasSize(2);
+    int expected = 1 - safeModeStates.get(0);
+    assertThat(expected).isEqualTo(safeModeStates.get(1));
+
+    // Check antenna states are valid.
+    assertThat(azimuthCommand).containsOnly(1.0);
```

Comment:
These monitoring events come every 1 sec. What I want to test here is all monitoring values are correct under situation where we don't know how many events come during the test.

Reply:
Ah, 'containsOnly' wasn't really clear to me (it means contains this value any number of times in any order).

So we do assume that at least one event is received.

Reply:
Yes, since telemetry comes every 10 sec and event comes every 1 sec, we will receive at lease 1 events (actually equals or more than 10 events) until the test receives two telemetry data.